pkg/platform/runtime/buildplan/buildplan.go

```diff
@@ -38,16 +131,53 @@ func NewMapFromBuildPlan(build *model.Build) (artifact.Map, error) {
 		}
 	}
 
-	for _, id := range terminalTargetIDs {
-		err := buildMap(id, lookup, res)
-		if err != nil {
-			return nil, errs.Wrap(err, "Could not build map for terminal %s", id)
+	if calculateBuildtimeClosure {
+		for _, id := range terminalTargetIDs {
+			if err := buildBuildtimeClosureMap(id, lookup, res); err != nil {
+				return nil, errs.Wrap(err, "Could not build map for terminal %s", id)
+			}
+		}
+	} else {
+		for _, id := range terminalTargetIDs {
+			if err := buildRuntimeClosureMap(id, lookup, res); err != nil {
+				return nil, errs.Wrap(err, "Could not build map for terminal %s", id)
+			}
```

Comment:
Nit: you could probably assign a `buildMap` variable to be the `buildBuildtimeClosureMap` or `buildRuntimeClosureMap` runction based on an `if calculateBuildtimeClosure` statement and just have a single loop that calls `buildMap(...)`.